test/partition.test.js

```diff
@@ -185,32 +186,60 @@ describe('cloudant - partitioned db', () => {
           },
         },
       });
-      db.automigrate('Product', () => {
-        Product.create(SEED_DATA, function(err) {
+      db.automigrate('Product', (err)=> {
+        if (err) return done(err);
+        Product.create(SEED_DATA, done);
+      });
+    });
+
+    it('find all records by partition key from option', (done) => {
+      Product.find({where: {tag: 'food'}}, {partitionKey: 'toronto'},
+        (err, results) => {
           if (err) return done(err);
-          Product.find({where: {tag: 'food'}}, {partitionKey: 'toronto'},
-            (err, results) => {
-              if (err) return done(err);
-              should.exist(results);
-              results.length.should.equal(2);
-              const resultTaggedFood = _.filter(results,
-                function(r) {
-                  return r.tag === 'food';
-                });
-              resultTaggedFood.length.should.equal(2);
-              done();
+          should.exist(results);
+          results.length.should.equal(2);
+          const resultTaggedFood = _.filter(results,
+            function(r) {
+              return r.tag === 'food' && r.city === 'toronto';
             });
+          resultTaggedFood.length.should.equal(2);
+          done();
+        });
+    });
+
+    it('find all records by partition key from property', (done) => {
+      Product.find({where: {tag: 'food', city: 'toronto'}},
+        (err, results) => {
+          if (err) return done(err);
+          should.exist(results);
+          results.length.should.equal(2);
+          const resultTaggedFood = _.filter(results,
+            function(r) {
+              return r.tag === 'food' && r.city === 'toronto';
+            });
+          resultTaggedFood.length.should.equal(2);
+          done();
+        });
+    });
+
+    it('partition key in options overrides property for find all', (done) => {
+      Product.find({where: {tag: 'food', city: 'toronto'}},
+        {partitionKey: 'london'}, (err, results) => {
+          if (err) return done(err);
+          should.exist(results);
+          results.length.should.equal(0);
```

Comment:
Do you mean the options overrides the query clause? 

And if it does, why isn't it result to 2? ( `food` and `london`)

Reply:
@agnes512 
> Do you mean the options overrides the query clause?

Yes, it's 0 because when search within partition `london`, there is no record with `city` as `toronto`. See the sample records created in https://github.com/strongloop/loopback-connector-cloudant/pull/230/files#diff-121c1d00e6380aa6109385e1df1ae803R237